### PR TITLE
[CoreData] Add support for Xcode 13 beta 3.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/coredata.cs [hunk 1]
```diff
@@ -1667,14 +1671,26 @@ interface NSPersistentHistoryTransaction : NSCopying
 	[DisableDefaultCtor] // NSInternalInconsistencyException Reason: NSCoreDataCoreSpotlightDelegate requires the use of the initializer initForStoreWithDescription:model: 
 	interface NSCoreDataCoreSpotlightDelegate
 	{
+
+		[Notification]
+		[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
```
Comment (spouliot) [inline src/coredata.cs:1676]: `[NoWatch]` and `[NoTV]` are already present on the type declaration and does not need to be repeated on members

File: src/coredata.cs [hunk 2]
```diff
@@ -1686,6 +1702,23 @@ interface NSCoreDataCoreSpotlightDelegate
...
+		[NoWatch, NoTV, Mac (12, 0), iOS (15, 0), MacCatalyst (15,0)]
+		[Export ("indexingEnabled")]
+		bool IndexingEnabled { [Bind ("isIndexingEnabled")] get; }
+
+		[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("startSpotlightIndexing")]
+		void StartSpotlightIndexing ();
+
+		[NoWatch, NoTV, Mac (12,0), iOS (15,0)]
+		[Export ("stopSpotlightIndexing"), MacCatalyst (15,0)]
```
Comment (spouliot) [inline src/coredata.cs:1720]: it's legit - but it's confusing to mix availability and `[Export]` inside the same `[...]`

File: src/coredata.cs [hunk 3]
```diff
@@ -2677,6 +2725,43 @@ interface NSPersistentCloudKitContainer {
 		[Watch (7,0), TV (14,0), Mac (11,0), iOS (14,0)]
 		[Export ("canModifyManagedObjectsInStore:")]
 		bool CanModifyManagedObjects (NSPersistentStore store);
+
+		// NSPersistentCloudKitContainer_Sharing
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerAcceptShareInvitationsResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("acceptShareInvitationsFromMetadata:intoPersistentStore:completion:")]
+		void AcceptShareInvitations (CKShareMetadata[] metadata, NSPersistentStore persistentStore, [NullAllowed] NSPersistentCloudKitContainerAcceptShareInvitationsHandler handler);
```
Comment (spouliot) [inline src/coredata.cs:2733]: the `NSPersistentCloudKitContainerAcceptShareInvitationsHandler` has `[NoWatch]` and `[NoTV]` attributes

File: src/coredata.cs [hunk 4]
```diff
@@ -2677,6 +2725,43 @@ interface NSPersistentCloudKitContainer {
...
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerPurgeObjectsAndRecordsInZone")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("purgeObjectsAndRecordsInZoneWithID:inPersistentStore:completion:")]
+		void PurgeObjectsAndRecordsInZone (CKRecordZoneID zoneId, [NullAllowed] NSPersistentStore persistentStore, [NullAllowed] NSPersistentCloudKitContainerPurgeObjectsAndRecordsInZoneHandler handler);
```
Comment (spouliot) [inline src/coredata.cs:2738]: same

File: src/coredata.cs [hunk 5]
```diff
@@ -2677,6 +2725,43 @@ interface NSPersistentCloudKitContainer {
...
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerPurgeObjectsAndRecordsInZone")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("purgeObjectsAndRecordsInZoneWithID:inPersistentStore:completion:")]
+		void PurgeObjectsAndRecordsInZone (CKRecordZoneID zoneId, [NullAllowed] NSPersistentStore persistentStore, [NullAllowed] NSPersistentCloudKitContainerPurgeObjectsAndRecordsInZoneHandler handler); 
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerPersistUpdatedShareResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("persistUpdatedShare:inPersistentStore:completion:")]
+		void PersistUpdatedShare (CKShare share, NSPersistentStore persistentStore, [NullAllowed] NSPersistentCloudKitContainerPersistUpdatedShareHandler handler);
```
Comment (spouliot) [inline src/coredata.cs:2743]: same

File: src/coredata.cs [hunk 6]
```diff
@@ -2677,6 +2725,43 @@ interface NSPersistentCloudKitContainer {
...
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerFetchParticipantsMatchingLookupInfosResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("fetchParticipantsMatchingLookupInfos:intoPersistentStore:completion:")]
+		void FetchParticipantsMatchingLookupInfos (CKUserIdentityLookupInfo[] lookupInfos, NSPersistentStore persistentStore, NSPersistentCloudKitContainerFetchParticipantsMatchingLookupInfosHandler handler);
```
Comment (spouliot) [inline src/coredata.cs:2748]: same

File: src/coredata.cs [hunk 7]
```diff
@@ -2677,6 +2725,43 @@ interface NSPersistentCloudKitContainer {
...
+
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("fetchSharesInPersistentStore:error:")]
+		[return: NullAllowed]
+		CKShare[] FetchSharesInPersistentStore ([NullAllowed] NSPersistentStore persistentStore, [NullAllowed] out NSError error);
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerShareManagedObjectsResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("shareManagedObjects:toShare:completion:")]
+		void ShareManagedObjects (NSManagedObject[] managedObjects, [NullAllowed] CKShare share, NSPersistentCloudKitContainerShareManagedObjectsHandler handler);
```
Comment (spouliot) [inline src/coredata.cs:2763]: same

**Review by spouliot — CHANGES_REQUESTED** [review]

**Review by rolfbjarne — APPROVED** [review]

File: src/coredata.cs (hunk 7 above)
Comment (mandel-macaque) [reply src/coredata.cs:2763]: All this, I got th first interface wrong and forgot to update the delegates, updated in https://github.com/xamarin/xamarin-macios/pull/12183/commits/a9c90671922a8fac77fa8580605bae1715fdf4ed

**Review by spouliot — APPROVED** [review]

File: src/coredata.cs [hunk 8]
```diff
@@ -1686,6 +1702,23 @@ interface NSCoreDataCoreSpotlightDelegate
...
+
+		[NoWatch, NoTV, Mac (12, 0), iOS (15, 0), MacCatalyst (15,0)]
+		[Export ("indexingEnabled")]
+		bool IndexingEnabled { [Bind ("isIndexingEnabled")] get; }
+
+		[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("startSpotlightIndexing")]
+		void StartSpotlightIndexing ();
+
+		[NoWatch, NoTV, Mac (12,0), iOS (15,0),MacCatalyst (15,0)]
```
Comment (tj-devel709) [inline src/coredata.cs:1719]:
```suggestion
		[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
```

**Review by tj-devel709 — APPROVED** [review]

**Pull request merged**